src/lib.rs

```diff
@@ -682,4 +682,29 @@ mod test {
             .collect();
         assert_eq!(scripthash_txs_txids, scripthash_txs_txids_async);
     }
+
+    #[cfg(all(feature = "blocking", any(feature = "async", feature = "async-https")))]
+    #[tokio::test]
+    async fn test_blocks() {
+        let (blocking_client, async_client) = setup_clients().await;
+        let start_height = BITCOIND.client.get_block_count().unwrap();
+        let blocks1 = blocking_client.blocks(None).unwrap();
+        let blocks_async1 = async_client.blocks(None).await.unwrap();
+        assert_eq!(blocks1, blocks_async1);
+        generate_blocks_and_wait(10);
+        let blocks2 = blocking_client.blocks(None).unwrap();
+        let blocks_async2 = async_client.blocks(None).await.unwrap();
+        assert_eq!(blocks2, blocks_async2);
+        assert_ne!(blocks2, blocks1);
+        let blocks3 = blocking_client.blocks(Some(start_height as u32)).unwrap();
+        let blocks_async3 = async_client
+            .blocks(Some(start_height as u32))
+            .await
+            .unwrap();
+        assert_eq!(blocks3, blocks_async3);
+        assert_eq!(blocks3, blocks1);
```

Comment:
Could check here that the returned blocks at `start_height` are actually the same we can receive via `BITCOIND.client`.

Reply:
I haven't seen this addressed.